app.py

```diff
@@ -1,14 +1,92 @@
 from flask import Flask, request, render_template
 import redis
+import psycopg2
+from psycopg2.extensions import AsIs
 import json
 import logging
 
 logger = logging.getLogger(__name__)
 
-r = redis.Redis(host='172.17.0.1', port=6379)
-
 app = Flask(__name__)
 
+def initialize():
+    global p
+    global r
+    try:
+        p = psycopg2.connect(user="postgres",
+                            password="postgres",
+                            host="172.17.0.1",
+                            port="5432",
+                            database="hello")
+
+        r = redis.Redis(host='172.17.0.1', port=6379)
+    except (Exception) as e:
+        return False
+    return True
+
+@app.route("/users", methods = ['POST', 'GET'])
+def users():
+
+    # If both are specified, id takes precedence
+    if request.method == 'GET':
+
+        c = p.cursor()
+
+        id   = request.args.get('id')
+        name = request.args.get('name')
+
+        try:
+            if id:
+                c.execute("SELECT * FROM users WHERE id = %s;", (id,))
+                record = c.fetchmany()
+            elif name and not id:
+                c.execute("SELECT * FROM users WHERE name = %s;", (name,))
+                record = c.fetchmany()
+            elif not id and not name:
+                c.execute("SELECT * FROM users")
+                record = c.fetchall()
+        except (Exception) as e:
+            return 'Could not connect to database', 503
+            exit(1)
+
+        if not record:
+            return_string = ""
+        else:
+            return_string = ""

```

Comment:
Instead of building your own json string here, I would recommend you drop everything into a dict, and then convert it to json using the builtin library.  This ensures you get valid json.
https://www.geeksforgeeks.org/how-to-convert-python-dictionary-to-json/